sns/data_mappings.bal

```diff
@@ -0,0 +1,335 @@
+    xml responseMeta = response/<namespace:ResponseMetadata>;
+    if (getSMSAttributesResponses.toString() != "") {
+        GetSMSAttributesResult getSMSAttributesResult = {
+            attributes: (getSMSAttributesResponses/<namespace:attributes>/*).toString()
+        };
+        ResponseMetadata responseMetadata = {
+            requestId: (responseMeta/<namespace:RequestId>/*).toString()
+        };
+        GetSMSAttributesResponse getSMSAttributesResponse = {
+            getSMSAttributesResult : getSMSAttributesResult,
+            responseMetadata : responseMetadata 
+        };
+        return getSMSAttributesResponse;
+    } else {
+        return error(response.toString());
+    }
+}
+
+isolated function xmlToGetSubscriprionAttributes(xml response) returns GetSubscriptionAttributesResponse|error {
+    xml getSubscriptionAttributesResponses = response/<namespace:GetSubscriptionAttributesResult>;
+    xml responseMeta = response/<namespace:ResponseMetadata>;
+    if (getSubscriptionAttributesResponses.toString() != "") {
+        GetSubscriptionAttributesResult getSubscriptionAttributesResult = {
+            attributes: (getSubscriptionAttributesResponses/<namespace:Attributes>/*).toString()
+        };
+        ResponseMetadata responseMetadata = {
+            requestId: (responseMeta/<namespace:RequestId>/*).toString()
+        };
+        GetSubscriptionAttributesResponse getSubscriptionAttributesResponse = {
+            getSubscriptionAttributesResult : getSubscriptionAttributesResult,
+            responseMetadata : responseMetadata 
+        };
+        return getSubscriptionAttributesResponse;
+    } else {
+        return error(response.toString());
+    }
+}
+
+isolated function xmlToConfirmedSubscriptionResponse(xml response) returns ConfirmedSubscriptionResponse|error {
+    xml confirmSubscriptionResponse = response/<namespace:ConfirmSubscriptionResult>;
+    xml responseMeta = response/<namespace:ResponseMetadata>;
+    if (confirmSubscriptionResponse.toString() != "") {
+        ConfirmedSubscriptionResult confirmSubscriptionResult = {
+            subscriptionArn : (confirmSubscriptionResponse/<namespace:SubscriptionArn>/*).toString()
+        };
+        ResponseMetadata responseMetadata = {
+            requestId: (responseMeta/<namespace:RequestId>/*).toString()
+        };
+        ConfirmedSubscriptionResponse confirmedSubscriptionResponse = {
+            confirmedSubscriptionResult : confirmSubscriptionResult,
+            responseMetadata : responseMetadata 
+        };
+        return confirmedSubscriptionResponse;
+    } else {
+        return error(response.toString());
+    }
+}
+
+isolated function xmlTopicListToTopicArray(xml response) returns string|error {
+    string|error topicName = (response/<namespace:TopicArn>/*).toString();
+    if (topicName is string) {
+        return topicName != "" ? topicName.toString() : "";
```

Comment:
`""` is used allover the code.

Shall we create a constant like `EMPTY_STRING` and reuse it?